src/lib/salad/rtree.h

```diff
@@ -53,24 +54,33 @@ typedef double area_t;
 #if defined(__cplusplus)
 extern "C" {
 #endif /* defined(__cplusplus) */
+/* A box in RTREE_DIMENSION space */
+
+enum {
+	/** Maximal possible R-tree height */
+	RTREE_MAX_HEIGHT = 16,
+	/** Maximal possible R-tree dimension */
+	RTREE_MAX_DIMENSION = 20
+};
+
+/* Coords */
+struct rtree_rect {
+	/* coords: { low X, upper X, low Y, upper Y, etc } */
+	coord_t coords[RTREE_MAX_DIMENSION * 2];
+};
 
 struct rtree_neighbor {
 	rb_node(struct rtree_neighbor) link;
 	struct rtree_neighbor *next;
 	void *child;
 	int level;
 	sq_coord_t distance;
+	/* Index. Is used for sorting neighbors with same distence */
+	struct index *index;
```

Comment:
As I already said, we are not allowed to use `index` here since it's another layer. Let's use a callback.